aries_vcx/src/handlers/connection/inviter/state_machine.rs

```diff
@@ -314,164 +313,119 @@ impl SmConnectionInviter {
-            }
+            _ => self.state
         };
         Ok(Self { state, ..self })
     }
 
     pub fn handle_send_response(self) -> VcxResult<Self> {
-        let Self { state, pairwise_info, send_message, .. } = self;
-        let state = match state {
+        let state = match self.state {
             InviterFullState::Requested(state) => {
-                match Self::_send_response(&state, &pairwise_info.pw_vk.clone(), send_message) {
-                    Ok(_) => {
-                        InviterFullState::Responded(state.into())
-                    }
+                match Self::_send_response(&state, &self.pairwise_info.pw_vk.clone(), self.send_message) {
+                    Ok(_) => InviterFullState::Responded(state.into()),
                     Err(err) => {
                         // todo: we should distinguish errors - probably should not send problem report
                         //       if we just lost internet connectivity
                         let problem_report = ProblemReport::create()
                             .set_problem_code(ProblemCode::RequestProcessingError)
                             .set_explain(err.to_string())
-                            .set_thread_id(&state.thread_id);
+                            .set_thread_id(&self.thread_id);
 
-                        send_message(&pairwise_info.pw_vk, &state.did_doc, &problem_report.to_a2a_message()).ok();
+                        (self.send_message)(&self.pairwise_info.pw_vk, &state.did_doc, &problem_report.to_a2a_message()).ok();
                         InviterFullState::Initial((state, problem_report).into())
                     }
                 }
             }
-            _ => state.clone()
+            _ => self.state
         };
-        Ok(Self { state, pairwise_info, ..self })
+        Ok(Self { state, ..self })
     }
 
     pub fn handle_ack(self, ack: Ack) -> VcxResult<Self> {
-        let Self { state, pairwise_info, send_message, .. } = self.clone();
-        let state = match state {
-            InviterFullState::Responded(state) => {
-                let thread_id = self.get_thread_id()?;
-                if !ack.from_thread(&thread_id) {
-                    let problem_report = ProblemReport::create()
-                        .set_problem_code(ProblemCode::RequestProcessingError)
-                        .set_explain(format!("Cannot handle Response: thread id does not match: {:?}", ack.thread))
-                        .set_thread_id(&thread_id); // TODO: Maybe set sender's thread id?
-
-                    send_message(&pairwise_info.pw_vk, &state.did_doc, &problem_report.to_a2a_message()).ok();
-                    InviterFullState::Initial((state, problem_report).into())
-                } else {
-                    InviterFullState::Completed((state, ack).into())
-                }
-            }
-            _ => {
-                state.clone()
-            }
+        let state = match self.state {
```

Comment:
we should still do `thread_id` verification, but the code should look simpler now I believe